### PR TITLE
Manachers algo Added #758

### PR DESCRIPTION
**Pull request by krishna-NIT**
#785 done in java
Manacher’s Algorithm – Linear Time Longest Palindromic Substring added
Please Review

### CODE REVIEW

File: Code/Java/longPalindrome.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+/*
+    Test Cases:
+        Input: abacdedcaba
+        Output: abacdedcaba
+
+        Input: abacdfgdcabba
+        Output: abba
+
+        Time Complexity: O(n)
```
Comment (rish-singhal) [inline Code/Java/longPalindrome.java:91]: describe what is n here

File: Code/Java/longPalindrome.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            }
+            // If palindrome centered at currentRightPosition i
+            // expand beyond centerRightPosition R,
+            // adjust centerPosition C based on expanded palindrome.
+            if (i + L[i] > R)
+            {
+                C = i;
+                R = i + L[i];
+            }
+        }
```
Comment (rish-singhal) [inline Code/Java/longPalindrome.java:75]: give appropriate line breaks to make the code more readable.

File: string/readme.md [hunk 3]
```diff
@@ -13,8 +13,9 @@ A string is a class that contains a char array, but automatically manages it for
 
 ## Questions :
 
-* Convert string to lowercase and uppercase ---->[C++](/Code/C++/string_to_lowercase_and_uppercase.cpp)
 * Revere a string using Stack ---->[C++](/Code/C++/reverse_a_string_using_stack.cpp)
-* Number of deletions to make a string pallindrome ---->[Python](/string/Number_of_deletions_to_make_pallindrome.py)
+* Manacher's algorithm (finding the longest palindrome in a string) ----> [Java](/Code/Java/longPalindrome.java)
```
Comment (rish-singhal) [inline string/readme.md:17]: Sort the list in alphabetical order

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: string/readme.md (hunk 3 above)
Comment (rish-singhal) [reply string/readme.md:17]: Update this*

File: Code/Java/longPalindrome.java (hunk 1 above)
Comment (rish-singhal) [reply Code/Java/longPalindrome.java:91]: This too

**Comment by rish-singhal** [comment]
Hey do look at the review comments

**Comment by rish-singhal** [comment]
Describe what n is.

**Comment by krishna-NIT** [comment]
Thanks for Reviewing it and Sorry for Delay from my side, @rish-singhal 
New Changes Done!

**Review by rish-singhal — APPROVED** [review]

**Comment by rish-singhal** [comment]
Great!

**Pull request merged**